### DISCUSSION TITLE
Search guide

### DISCUSSION THREAD

**Discussion by andrejohansson** [immich-app/immich#3657] [Q&A]
I have looked for a search guide but haven´t quite found answers to my question, and the question is: 

- Is it possible to search specifically using metadata?
- Can I save searches (eg, an album should actually be a dynamic search question?

Examples of queries I want to do:

- Find all photos between date x and y taken with camera z with quality over 20MP
- Find all photos containing person x between date x and y
- Find all photos having metadata field z (exact match)
- Find multiple photos by a list of filenames, like `m:filename.jpg`
- Using operators in search query, like `AND` and `OR`

Thank you for a wonderful project!

**Comment by kaizh102** [comment]
such a clarification is badly needed.

**Comment by zodcas** [comment]
Adding this one too.
- Find all photos containing person x and person y in the same photo.

**Reply by pentix** [reply]
Came here because of this. This was previously possible by using a search term like `m:personA m:personB`, but I think with the upgrade to 1.91 this functionality was lost.

**Reply by troyane** [reply]
I'm also here to "Find all photos containing person x and person y in the same photo." I would like to know how this can be done. Thanks.

**Comment by jrasm91** [comment]
It is not yet, hopefully do soon ish.

**Comment by davidpan** [comment]
link: https://github.com/immich-app/immich/pull/4931

**Comment by rantanlan** [comment]
Definitely needed, right now my only chance to find a specific picture is to add it to an album. A search for an image description or a tag I'll add to these specific pictures would be highly appreciated.

**Comment by aisbergde** [comment]
for doing this kind of search I am using digiKam. It would be great, if some of these searches would be possible to do in immich.

**Comment by Avatar007z** [comment]
+1 for this feature

**Comment by lythandas** [comment]
Immich really needs to strenghten its searching features imo. Searching photo by date using only the date sliders on the right is tedious.

**Comment by dragere** [comment]
Also
- is it possible to manually set tags and search for them?
- how to search for descriptions
